base/doc/usrguide.tex

```diff
@@ -860,6 +860,57 @@ \subsection{Typesetting verbatim-like material}
 Similarly, the \texttt{verbatim} environment sets up the meaning of \cs{par}
 suitable for breaking lines.
 
+\subsection{Verbatim environments}
+\label{sec:cmd:verbenv}
+
+In some cases, as well as grabbing an environment body you will want the
+contents to be treated verbatim. This is available using the argument
+specification~\texttt{f}. Like the \texttt{b} specification, this has to be the
+last one. Thus for example
+\begin{verbatim}
+\NewDocumentEnvironment{MyVerbatim}{!O{\ttfamily} f}
+  {\begin{center} #1 #2\end{center}} {}
+\begin{MyVerbatim}[\ttfamily\itshape]
+  % Some code is shown here
+  $y = mx + c$
+\end{MyVerbatim}
+\end{verbatim}
+will typeset
+\NewDocumentEnvironment{MyVerbatim}{!O{\ttfamily} f}
+  {\begin{center} #1 #2\end{center}} {}
+\begin{MyVerbatim}[\ttfamily\itshape]
+  % Some code is shown here
+  $y = mx + c$
+\end{MyVerbatim}
+
+As grabbing the entire contents verbatim means there are no \cs{par} tokens,
+newlines are always permitted: there is no need for a |+| here. As for the
+\texttt{v} specification, newlines are stored as \cs{obeyedline}. In a similar
+fashion to the \texttt{b}~specification, by default \emph{newlines} are trimmed
+at both ends of the body. Putting the prefix |!| before \texttt{f} suppresses
+space-trimming.
+
+Collection of the body takes place on a line-by-line basis: content is
+collected up to the end-of-line in the source, then examined before storage.
+This means that the line ending the environment (containing in the example
+above |\end{MyVerbatim}|) cannot have any text \emph{after} the end of the
+environment. Text \emph{before} the end of environment is treated normally,
+but note that there is no trailing \cs{obeyedline} added if there is text here.
+Other than optional arguments, no text is allowed on the opening line of the
+environment.
+
+Special handling is applied to a \texttt{o}, \texttt{O}, \texttt{d} or
+\texttt{D}~specification argument immediately before an
+\texttt{f}~specification. This means that when the optional argument is absent,
+the first character of the next line will be read properly. Issues may arise if
+\emph{multiple} optional arguments are used before an \texttt{f}~specification,
+and are therefore strongly discouraged.
+
+Notice that for technical reasons, we recommend that an optional argument
+coming immediately before an \texttt{f} specification should not allow any
+spaces, achieved by adding the \texttt{!} as showing in the example. However,
```

Comment:
not clear which spaces are meant (suggest to be explicit and say ignore spaces while scannign for the optional argument, or something like that).

Reply:
now I dea why this showed up twice ... must have been a hickup with my connection